controllers/multiclusterobservability/multiclusterobservability_controller.go

```diff
@@ -447,12 +456,45 @@ func (r *MultiClusterObservabilityReconciler) SetupWithManager(mgr ctrl.Manager)
 		Owns(&observatoriumv1alpha1.Observatorium{}).
 		// Watch the configmap for thanos-ruler-custom-rules update
 		Watches(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForObject{}, builder.WithPredicates(cmPred)).
-		// Watch the configmap for mch-image-manifest-* update
-		Watches(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForObject{}, builder.WithPredicates(imageManifestPred)).
 		// Watch the secret for deleting event of alertmanager-config
-		Watches(&source.Kind{Type: &corev1.Secret{}}, &handler.EnqueueRequestForObject{}, builder.WithPredicates(secretPred)).
-		// actually create the controller with the reconciler
-		Complete(r)
+		Watches(&source.Kind{Type: &corev1.Secret{}}, &handler.EnqueueRequestForObject{}, builder.WithPredicates(secretPred))
```

Comment:
so we handle 2 cases here:
1. if there is no mch exists, then we just ignore the watch. it applies to instance observability on top of server foundation.
2. for ACM product, we assume there is always MCH exists before starting observability service.
Correct?

Reply:
Correct, we have to check if the MCH CRD existing before adding acheme and filtered cache.